src/zoid/buttons/component.jsx

```diff
@@ -716,11 +716,17 @@ export const getButtonsComponent: () => ButtonsComponent = memoize(() => {
             messageType,
             offerCountryCode,
             creditProductIdentifier,
+            merchantID,
           }) => {
-            const { message, clientID, merchantID, currency, buttonSessionID } =
-              props;
+            const { message, clientID, currency, buttonSessionID } = props;
             const amount = message?.amount;
 
+            // since we are adding our own merchantID
+            // this should override the default coming from sdk client
+            getLogger().addTrackingBuilder(() => ({
+              [FPTI_KEY.SELLER_ID]: merchantID,
+            }));
```

Comment:
What happens when an actual `merchantID` is passed in as part of the SDK script (i.e. partner integration)? Will this unintentionally override that or are we still good there? Also will this impact other non-messaging events that are fired from outside the IFrame since `addTrackingBuilder` changes the shared logger instance?

Reply:
let me attempt that now